internal/tasks/repotester/repotester.go

```diff
@@ -59,6 +73,100 @@ func (r *RepoTester) TestRepositories() {
 	}
 }
 
+func configureRepoHTTPSClient(httpConfig api.RepositoryHttpConfig) error {
+	tlsConfig := tls.Config{} //nolint:gosec
+	if httpConfig.SkipServerVerification != nil {
+		tlsConfig.InsecureSkipVerify = *httpConfig.SkipServerVerification //nolint:gosec
+	}
+
+	if httpConfig.TlsClientCertData != nil && httpConfig.TlsClientCertKey != nil {
+		cert, err := b64.StdEncoding.DecodeString(*httpConfig.TlsClientCertData)
+		if err != nil {
+			return err
+		}
+
+		key, err := b64.StdEncoding.DecodeString(*httpConfig.TlsClientCertKey)
+		if err != nil {
+			return err
+		}
+
+		tlsPair, err := tls.X509KeyPair(cert, key)
+		if err != nil {
+			return err
+		}
+
+		tlsConfig.Certificates = []tls.Certificate{tlsPair}
+	}
+
+	if httpConfig.RootCA != nil {
+		ca, err := b64.StdEncoding.DecodeString(*httpConfig.RootCA)
+		if err != nil {
+			return err
+		}
+
+		rootCAs, _ := x509.SystemCertPool()
+		if rootCAs == nil {
+			rootCAs = x509.NewCertPool()
+		}
+		rootCAs.AppendCertsFromPEM(ca)
+		tlsConfig.RootCAs = rootCAs
+	}
+
+	gitclient.InstallProtocol("https", githttp.NewClient(
+		&http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tlsConfig,
+			},
+		},
+	))
+	return nil
+}
+
+func GetAuth(repository *model.Repository) (transport.AuthMethod, error) {
+	if httpRepoRegExp.MatchString(*repository.Spec.Data.Repo) && repository.Spec.Data.HttpConfig != nil {
+		if strings.HasPrefix(*repository.Spec.Data.Repo, "https") {
+			err := configureRepoHTTPSClient(*repository.Spec.Data.HttpConfig)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		if repository.Spec.Data.HttpConfig.Username != nil && repository.Spec.Data.HttpConfig.Password != nil {
+			auth := &githttp.BasicAuth{
+				Username: *repository.Spec.Data.HttpConfig.Username,
+				Password: *repository.Spec.Data.HttpConfig.Password,
+			}
+			return auth, nil
+		}
+	} else if scpLikeUrlRegExp.MatchString(*repository.Spec.Data.Repo) {
+		if repository.Spec.Data.SshConfig != nil {
+
+			sshPrivateKey, err := b64.StdEncoding.DecodeString(*repository.Spec.Data.SshConfig.SshPrivateKey)
+			if err != nil {
+				return nil, err
+			}
+
+			repoSubmatch := scpLikeUrlRegExp.FindStringSubmatch(*repository.Spec.Data.Repo)
+			password := ""
+			if repository.Spec.Data.SshConfig.PrivateKeyPassphrase != nil {
+				password = *repository.Spec.Data.SshConfig.PrivateKeyPassphrase
+			}
+			auth, err := gitssh.NewPublicKeys(repoSubmatch[1], sshPrivateKey, password)
+			if err != nil {
+				return nil, err
+			}
+			if repository.Spec.Data.SshConfig.SkipServerVerification != nil && *repository.Spec.Data.SshConfig.SkipServerVerification {
+				auth.HostKeyCallbackHelper = gitssh.HostKeyCallbackHelper{
+					HostKeyCallback: ssh.InsecureIgnoreHostKey(), //nolint:gosec
+				}
+			}
+
+			return auth, nil
+		}
+	}
+	return nil, nil
```

Comment:
Should we document that finding no auth returns a nil AuthMethod? Or is this  an error?

Reply:
`nil` is not an error in this case. It simply means that repo did not have any ssh/http config. 
It is documented now.